components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -377,19 +377,37 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
         this.setState({ workspaceInstance, error });
     }
 
-    async ensureWorkspaceAuth(instanceID: string) {
+    async ensureWorkspaceAuth(instanceID: string, retry: boolean) {
         if (!document.cookie.includes(`${instanceID}_owner_`)) {
-            const authURL = gitpodHostUrl.asWorkspaceAuth(instanceID);
-            const response = await fetch(authURL.toString());
-            if (response.redirected) {
-                this.redirectTo(response.url);
-                return;
-            }
-            if (!response.ok) {
-                // getting workspace auth didn't work as planned - redirect
-                this.redirectTo(authURL.asWorkspaceAuth(instanceID, true).toString());
+            let attempt = 0;
+            while (attempt <= 10) {
+                attempt++;
+
+                const authURL = gitpodHostUrl.asWorkspaceAuth(instanceID);
+                const response = await fetch(authURL.toString());
+                if (response.status === 404 && retry) {
+                    console.warn("Unable to retrieve workspace-auth cookie! Retrying shortly...", {
+                        code: response.status,
+                        attempt,
+                    });
+                    // If the token is not there, we assume it will appear, soon: Retry a couple of times.
+                    await new Promise((resolve) => setTimeout(resolve, 2000));
+                    continue;
+                }
+                if (!response.ok) {
+                    // getting workspace auth didn't work as planned
+                    console.error("Unable to retrieve workspace-auth cookie! Quitting.", {
+                        code: response.status,
+                        attempt,
+                    });
+                    return;
+                }
+
+                // Response code is 200 at this point: done!
+                console.info("Retrieved workspace-auth cookie.", { code: response.status, attempt });
                 return;
             }
+            console.error("Unable to retrieve workspace-auth cookie! Giving up.", { attempt });
```

Comment:
This would always print `attempt: 10`, no?

Reply:
Jup! I feel that's ok.

BTW: I realized after pushing this that it won't improve our metrics/perception because we don't have any insight into client-side errors. :cry: ATM we can assume that every entry we have in the server logs equals an unhappy user, but with the retries that does not have to be the case anymore.
But changing that is definitely a follow-up PR. :lotus_position: